public/javascripts/signup/signup.js

```diff
@@ -0,0 +1,78 @@
+(function(){
+	//initialize the angular app and inject dependencies.
+	angular.module("olinbaja.signup", ['ngRoute'])
+    .config(function($routeProvider, $locationProvider) {
+      $routeProvider
+        .when('/signup', {
+          templateUrl: './pages/signup.html',
+          controller: 'SignupController',
+          controllerAs: 'vm'
+        })
+
+      $locationProvider.html5Mode({
+        enabled: true,
+        requireBase: false
+      });
+    })
+    .controller('SignupController', SignupController);
+
+  function SignupController($http, $window) {
+    var vm = this;
```

Comment:
https://johnpapa.net/angularjss-controller-as-and-the-vm-variable/

It's a best practice. It represents ViewModel. It is clean, short, and expressive, and at the same time removes any binding issues with `this` (i.e. accidentally closing over `this`).

It also makes it cleaner to read in all other code (especially HTML). It